include/dca/function/function.hpp

```diff
@@ -185,16 +185,12 @@ class function {
   void operator*=(const function<scalartype, domain>& other);
   void operator/=(const function<scalartype, domain>& other);
 
-  template <typename new_scalartype>
-  void operator=(new_scalartype c);
-  template <typename new_scalartype>
-  void operator+=(new_scalartype c);
-  template <typename new_scalartype>
-  void operator-=(new_scalartype c);
-  template <typename new_scalartype>
-  void operator*=(new_scalartype c);
-  template <typename new_scalartype>
-  void operator/=(new_scalartype c);
+
+  void operator=(scalartype c);
+  void operator+=(scalartype c);
+  void operator-=(scalartype c);
+  void operator*=(scalartype c);
+  void operator/=(scalartype c);
```

Comment:
Avoid matching the scalar assignment overload to the function assignment of a non-const function. It also removes one copy.